Add Options to scheduler stress

chaptersix opened the pull request:
Rework the scheduler_stress scenario for flexible ad-hoc load testing.

Changes:
- Add `skip-deletion`, `skip-creation`, and `worker-count` options for controlling load patterns without rebuilding
- Refactor execution into worker pool with channel-based schedule dispatch (configurable concurrency via `worker-count`)
- Add `executeWithExistingSchedules` path to re-run operations against previously created schedules (`skip-creation=true`)
- Add `retryWithBackoff` with exponential backoff and jitter for all schedule operations (create, describe, update, delete)
- Per-operation 30s timeouts to avoid hanging on unresponsive calls
- Set `TriggerImmediately` on schedule creation
- Debug logging for schedule deletion lifecycle

chaptersix committed:
first v2 load test

chaptersix committed:
feat: add debug logging for schedule deletion lifecycle

chaptersix committed:
Merge branch 'main' into alex/sch-adj

lina-temporal commented on scenarios/scheduler_stress.go:
```diff
@@ -164,53 +182,202 @@ func (s *SchedulerExecutor) Execute(ctx context.Context, run *loadgen.Run) error
+		wg.Go(func() {
+			// Each worker keeps consuming from the channel until it's closed
+			for config := range scheduleConfigChan {
+				func() {
```
I don't think this needs to be in a lambda, given that it's already in the `wg.Go(func() { ... })` scope

chaptersix committed:
refactor: address PR review feedback for scheduler stress

- Remove unnecessary lambdas and defers, use sequential flow instead
- Add ctx.Done select to all ticker waits via waitForTick helper
- Extract shared deletion logic into waitAndDeleteSchedule
- Move channel close responsibility into listSchedules
- Fix stale retryConfig comment
- Document why custom retry is needed (SDK skips DeadlineExceeded)

chaptersix committed:
Merge branch 'main' into alex/sch-adj
